### PR TITLE
BugFix : Fixed Improper Handling for Non-Numeric Values in Allotted hours section of Create Action item modal

### LINKED ISSUE #3368: Bug Report : Improper input handling for Allotted hours in the Create Action Item modal
**Describe the bug**
When entering non-numeric characters in the allotted hours input field, the field value becomes NaN (Not a Number) and remains stuck in this state. Once NaN appears, the field cannot be cleared or updated with new values, even when attempting to backspace or enter valid numbers. This creates a blocking user experience where the only way to recover is to refresh the form.
To Reproduce

**Steps to reproduce the behavior:**
1. Login as a admin or superadmin
2. Navigate to action items 
3. Try to create a action item
4. type a letter in the allotted hours section.

**Expected behavior**

1.When non-numeric characters are entered, the field should either prevent the input or maintain its previous valid value
2.Users should be able to clear the field and enter new values at any time
3.The field should never display NaN to end users
4.Invalid inputs should be handled gracefully without getting the field stuck in an unrecoverable state

**Actual behavior**

1.Entering a letter or non-numeric character causes the field value to become NaN
2.Once NaN appears, the field becomes unresponsive to new inputs
3.Backspace and delete operations fail to clear the NaN value
4.The field remains stuck in an invalid state until form refresh

**Screenshots**

Image

**Additional details**
Root cause appears to be in the handleFormChange function where NaN values are being stored in the form state without proper validation. The current implementation doesn't properly sanitize numeric inputs before updating the state, leading to this persistent error state.

I have resolved this error on  my local machine and am ready to raise a PR for this.

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by yugal07**
**What kind of change does this PR introduce?**

BugFix

**Issue Number:**

Fixes #3368 

**Snapshots/Videos:**

Before 

https://github.com/user-attachments/assets/4fd6a4ce-ef56-4a21-a1a2-c8729ed7a9c1

After

https://github.com/user-attachments/assets/486f11f1-336a-48c6-a544-3f4bdd48d9c0

**If relevant, did you update the documentation?**

N/A

**Summary**

Added separate handling for allotted hours section, for non numeric values and negative values

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced validation for allotted hours input to ensure only valid, non-negative numbers are accepted.
	- Added test cases to validate that the allotted hours field does not accept letters or negative values.
	- Added test case to ensure that the due date input field correctly resets when assigned a null value.
	- Introduced additional test cases to cover edge cases for allotted hours input.
- **New Features**
	- Introduced a new `ProfileDropdown` component in the sidebar for user profile management.
	- Updated styling for various components to improve visual consistency and responsiveness.
- **Chores**
	- Removed redundant CSS files and updated import paths to a centralized stylesheet for better maintainability.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Create a test for this so that it never happens again.
2. Also, ensure that you are using this modal.

https://github.com/PalisadoesFoundation/talawa-admin/pull/2984#issuecomment-2585396521

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please ensure Code Rabbit approves your PR

**Comment by yugal07** [comment]
@palisadoes, Sure. will make the necessary changes add the commit

**Comment by yugal07** [comment]
@palisadoes, I've added the test cases and implemented Code Rabbit's suggestions. Let me know if any further adjustments are needed.

**Comment by palisadoes** [comment]
Please fix the failing test

**Comment by yugal07** [comment]
@palisadoes, The tests are failing due to the presence of code coverage disable lines in the file. However, I did not add them; they were already there.

**Comment by palisadoes** [comment]
Please remove them and add the appropriate tests

**Comment by yugal07** [comment]
@coderabbitai full review

File: src/screens/OrganizationActionItems/ItemModal.tsx [hunk 1]
```diff
@@ -226,6 +226,16 @@ const ItemModal: FC<InterfaceItemModalProps> = ({
...
+    // Special handling for allottedHours
+    if (field === 'allottedHours') {
+      // If the value is not a valid number or is negative, set to null
+      const numValue = typeof value === 'string' ? Number(value) : value;
+      if (typeof numValue !== 'number' || isNaN(numValue) || numValue < 0) {
+        setFormState((prevState) => ({ ...prevState, [field]: null }));
+        return;
+      }
+    }
+
```
[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.tsx]: Done

[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.tsx]: It loses functionality if i add all your code. I have replaced isNaN with Number.isNaN

File: src/screens/OrganizationActionItems/ItemModal.tsx [hunk 2]
```diff
@@ -226,6 +226,20 @@
...
+      const numValue = typeof value === 'string' ? Number(value) : value;
+      if (
+        typeof numValue !== 'number' ||
+        Number.isNaN(numValue) ||
+        numValue < 0
+      ) {
+        setFormState((prevState) => ({ ...prevState, [field]: null }));
+        return;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.tsx:241]: Done

[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.tsx:241]: I have added that case from line 819 to 821

**Comment by yugal07** [comment]
@coderabbitai full review

**Comment by yugal07** [comment]
@palisadoes, I've made the necessary changes. Let me know if anything else needs to be adjusted. Thanks!

**Comment by palisadoes** [comment]
1. Please make sure code rabbit ai approved your changes
1. Try to get the code coverage for your patch to 100%

**Comment by yugal07** [comment]
There was an issue with the head branch. When I rebased to fix conflicts, it pulled in other contributors' commits too. I'll create a new branch and open a fresh PR to fix this.
